.github/workflows/deploy-ghpages-latest-stable.yml

```diff
@@ -128,12 +128,12 @@ jobs:
         ls stable # TODO: remove this debugging line
     # TODO: use the matrix of versions we want to implement 
     - name: Upload latest artifact
-      uses: actions/upload-artifact@v1
+      uses: actions/upload-pages-artifact@v1
       with:
         name: latest
         path: new_docs/latest
     - name: Upload stable artifact
-      uses: actions/upload-artifact@v1
+      uses: actions/upload-pages-artifact@v1
```

Comment:
None of the two should be uploaded. There has to be a single `actions/upload-pages-artifact` in the whole workflow.

In practice, you should remove both of them, and upload a single one, i.e. `path: new_docs`, and that's it.
If you need the `index.html`, add the redirect I already mentioned.